Merge Dark Sky docs

NeLLyMerC opened the pull request:
Attempting to merge Dark Sky documentation for simplification and clarity. If the PR is accepted, I'd like to remove /integrations/weather.darksky/ and just keep the current page (/integrations/darksky/)

## Proposed change
<!-- 
  The Dark Sky documents are split and not easy for a beginner to understand. I want to merge the documentation and provide greater detail, which would provide simplification and clarity. I am unsure how to add a screenshot, but that is something I want to include as well.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
  I want to additionally delete the weather.darksky link if this PR is merged. I may need an example on how to add images for the final version of this page.
-->

- Link to parent pull request in the codebase:
- This PR fixes or closes issue:

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [ x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

NeLLyMerC committed:
Merge Dark Sky docs

Attempting to merge Dark Sky documentation for simplification and clarity. If the PR is accepted, I'd like to remove /integrations/weather.darksky/ and just keep the current page (/integrations/darksky/)

klaasnicolaas commented:
We would prefer that you now immediately remove the old platform page and set the redirects correctly from weather.darksky to darksky.

klaasnicolaas commented on source/_integrations/darksky.markdown:
```diff
@@ -235,6 +237,48 @@ All language options are described in this table that you can use for the dark s
+  type: string
+  default: hourly
+{% endconfiguration %}
+
```
Remove the extra whitespace

NeLLyMerC replied:
Whitespace removed, thank you.

klaasnicolaas commented on source/_integrations/darksky.markdown:
```diff
@@ -235,6 +237,48 @@ All language options are described in this table that you can use for the dark s
 While the platform is called "darksky" the sensors will show up in Home Assistant as "dark_sky" (eg: sensor.dark_sky_summary).
-</div>
+</div>  
+
```
Maybe add here a header with **Weather**

klaasnicolaas commented on source/_integrations/darksky.markdown:
```diff
@@ -24,7 +24,9 @@ You need an API key which is free but requires [registration](https://darksky.ne
 
-To add Dark Sky to your installation, add the following to your `configuration.yaml` file:
+There are two ways you can integrate Dark Sky into Home Assistant: as a sensor and a weather entity. The [sensor](/integrations/sensor) entity allows you to add the conditions you'd like to monitor and returns a value that you can add to various styles of Lovelace cards. In contrast, the weather entity allows you to set up a [weather forecast card](/lovelace/weather-forecast/#configuration-variab ...
+
```
Maybe add here a header with **Sensor**

NeLLyMerC replied:
Added, thank you for the feedback

NeLLyMerC committed:
Update darksky.markdown

- Added headers for newly added sections of “Sensor” and “Weather”
- Removed extra whitespace

NeLLyMerC reviewed (commented):
Edited document based on feedback

frenck commented:
These are 2 different things in Home Assistant. How would one know the difference know? Where is it explained?

NeLLyMerC commented:
> These are 2 different things in Home Assistant. How would one know the difference know? Where is it explained? And how would that go with updates?
> 
> I think we should keep things that are separate in our code base, separate in the documentation as well. Mainly since the documentation should reflect the codebase.

The differences are described in one of the first paragraphs on the edits I made. Each entity type is linked to a page for reference (/integration/sensors & /lovelace/weather-forecast/), so the user can see examples of the differences. There are also two separate headings with the example configuration needed to get each added to Home Assistant. I found it confusing as it is because searching “Dark Sky” in the integrations page yields two results that say the same thing. The current page of /integrations/weather.darksky is also not very descriptive of what you will gain by adding this type of integration.

frenck commented:
I'm assigning myself as well, I think having weather platforms both implemented as weather and sensor actually are unfriendly. I'm going to take up this discussion on that level first.

NeLLyMerC commented:
The idea I was going for is similar to other integrations like Canary, Bloomsky, or Arlo. One page for the integration that shows you the different entities that can be configured for it. 
Let me know how I can help

frenck commented:
Yes, I get that, however, in this case it are 2 different integrations/platforms. (different source).

frenck commented:
Not stale, thanks stable bot.
